packages/select-rich/src/LionSelectInvoker.js

```diff
@@ -9,11 +9,10 @@ import { css, html } from '@lion/core';
 /**
  * LionSelectInvoker: invoker button consuming a selected element
  */
-// @ts-expect-error
 export class LionSelectInvoker extends LionButton {
   static get styles() {
     return [
-      super.styles,
+      ...super.styles,
```

Comment:
So basically we should make sure we also use `CSSResult[]` inside our components, no matter what, but it would be nice if LitElement would enforce this in the next version.
Doesn't the example below result in a type of `Array<CSSResult|CSSResult[]>` ?

Shouldn't we put `...(super.styles || [])` everywhere?

Shouldnt

Reply:
> Doesn't the example below (Tooltip) result in a type of Array<CSSResult|CSSResult[]> ?

Yep. But LitElement is smart enough to merge/flatten deeply nested arrays.. 

I explained in more detail in Mattermost team chat ;) it's a little complicated.. definitely agree Lit should make this consistent and enforce, but I'm a bit concerned it will go a bit beyond their breaking change budget for v3

Reply:
I know LitElement allows, it even would allow `Array<CSSResult|Array<CSSResult|CSSResult[]>>` etc.
But if we force all implementation in our code base to only use `CSSResult[]`, wouldn't it be solved?

Reply:
Hmm but `super.styles` is not always iterable. If you extend from LitElement like we do in LionTooltip, it can also be `CSSResult` or `CSSStyleSheet` instead of `CSSResultArray`. So in order to always get our own styles to use `CSSResult[]` everywhere, we need to typecast the super styles in a bunch of places to only "allow" `CSSResult[]`. I'm fine with doing that to get things consistent, maybe create an issue?